### PR TITLE
CIRC-1140 - No Log Record entry when loan due date was changed

### PR DESCRIPTION
**Pull request by khandramai**
[CIRC-1140](https://issues.folio.org/browse/CIRC-1140) - No Log Record entry when loan due date was changed

## Purpose
No Log Record entry when loan change due date was made.

## Approach
Enable publishing of due date changes, improve event description

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Review by alekGbuz — APPROVED** [review]

**Review by siarhei-charniak — APPROVED** [review]

**Review by obozhko-folio — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -33,6 +33,7 @@
     <lombok.version>1.18.16</lombok.version>
     <spring.version>5.2.7.RELEASE</spring.version>
     <maven.site.plugin.version>3.9.1</maven.site.plugin.version>
+    <sonar.exclusions>src/main/java/org/folio/circulation/domain/LoanAndRelatedRecords.java</sonar.exclusions>
```
Comment (marcjohnson-kint) [inline pom.xml:36]: Why has this been excluded from Sonar's checking?

File: pom.xml [hunk 2]
```diff
@@ -206,6 +207,11 @@
       <version>${rmb.version}</version>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>org.jmockit</groupId>
+      <artifactId>jmockit</artifactId>
+      <version>1.49</version>
+    </dependency>
```
Comment (marcjohnson-kint) [inline pom.xml:214]: I thought this project already included mockito, does jmockit do something different?

Comment (khandramai) [reply pom.xml]: Hi @marc-johnson , yes, I 've deleted jmockit and reused existing approach.

File: pom.xml (hunk 1 above)
Comment (khandramai) [reply pom.xml]: I removed this exclusion. The problem with code coverage comes from using jmockit javaagent. It is automatically fixed when I deleted jmockit.

**Pull request merged**